src/lob-address-elements.js

```diff
@@ -1,527 +1,724 @@
-                        primary_line: data.primary_line
-                    }
-                }
-            }
+        function hideMessages() {
+          config.elements.message.hide();
+          config.elements.primaryMsg.hide();
+          config.elements.secondaryMsg.hide();
+          config.elements.cityMsg.hide();
+          config.elements.stateMsg.hide();
+          config.elements.zipMsg.hide();
+        }
 
-            /**
-             * Called after Lob verifies an address. Improves/updates and caches
-             * @param {object} data - verified address object returned from Lob
-             */
-            function fixAndSave(data) {
-                var didFix;
-                var parts = denormalizeParts(data, !!config.elements.secondary.val());
-                var address = config.address = {
-                    primary: parts.primary_line,
-                    secondary: parts.secondary_line,
-                    city: data.components && data.components.city || '',
-                    state: data.components && data.components.state || '',
-                    zip: plus4(data.components)
-                }
-                for (var p in address) {
-                    if (address.hasOwnProperty(p)) {
-                        if (address[p].toUpperCase() != config.elements[p].val().toUpperCase() &&
-                            !(p === 'zip' && config.elements[p].val().length === 5 && address[p].indexOf(config.elements[p].val()) == 0)) {
-                            didFix = true;
-                        }
-                        config.elements[p].val(address[p]);
-                    }
-                }
-                return didFix;
-            }
+        /**
+         * Show form- and field-level error messages as cofigured. Verification did NOT succeed.
+         * @param {object} err - Verification error object representing a Lob error type
+         * @param {string} err.msg - Top-level message to show for the form
+         * @param {string} err.type - Specific error type to apply at field level if relevant
+         */
+        config.do.message = function (err) {
+          if (err) {
+            config.elements.message.text(err.msg).show('slow');
+            messageTypes[err.type] && messageTypes[err.type](config.messages[err.type]);
+          }
+        }
+
+        var messageTypes = {
+          primary_line: function (msg) {
+            config.elements.primaryMsg.text(msg).show('slow');
+          },
+          city_state_zip: function (msg) {
+            config.elements.cityMsg.text(msg).show('slow');
+            config.elements.stateMsg.text(msg).show('slow');
+            config.elements.zipMsg.text(msg).show('slow');
+          },
+          zip: function (msg) {
+            config.elements.zipMsg.text(msg).show('slow');
+          },
+          deliverable_missing_unit: function (msg) {
+            (config.denormalize === false &&
```

Comment:
Is there a reason we're not using `!config.denormalize`

Reply:
good catch. fixed in 3 other places as well.